inline-c-cpp/src/Language/C/Inline/Cpp/Exceptions.hs

```diff
@@ -161,22 +161,7 @@ exceptionalValue typeStr =
 tryBlockQuoteExp :: String -> Q Exp
 tryBlockQuoteExp blockStr = do
   let (ty, body) = C.splitTypedC blockStr
-  _ <- C.include "<exception>"
-  _ <- C.include "<cstring>"
-  _ <- C.include "<cstdlib>"
   _ <- C.include "HaskellException.hxx"
-  -- see
-  -- <https://stackoverflow.com/questions/28166565/detect-gcc-as-opposed-to-msvc-clang-with-macro>
-  -- regarding how to detect g++ or clang.
-  --
-  -- the defined(__clang__) should actually be redundant, since apparently it also
-  -- defines GNUC, but but let's be safe.
-  _ <- C.verbatim $ unlines
-    [ "#if defined(__GNUC__) || defined(__clang__)"
-    , "#include <cxxabi.h>"
-    , "#include <string>"
-    , "#endif"
-    ]
   typePtrVarName <- newName "exTypePtr"
   msgPtrVarName <- newName "msgPtr"
   haskellExPtrVarName <- newName "haskellExPtr"
```

Comment:
The comment below:

```
  -- see
  -- <https://stackoverflow.com/questions/561997/determining-exception-type-after-the-exception-is-caught/47164539#47164539>
  -- regarding how to show the type of an exception.
```

should be moved to the factored out functions that you have defined.

Reply:
@bitonic  Thank you for your  reviewing.
I've move the comment and add  a function of currentExceptionTypeName showing the exception-name.